tests/integration/security/util/reachability/context.go

```diff
@@ -153,7 +169,7 @@ func (rc *Context) Run(testCases []TestCase) {
 			ctx.Logf("[%s] [%v] Finish waiting. Continue testing.", testName, time.Now())
 
 			for _, src := range []echo.Instance{rc.A, rc.B, rc.Headless, rc.Naked} {
-				for _, dest := range []echo.Instance{rc.A, rc.B, rc.Headless, rc.Multiversion, rc.Naked} {
+				for _, dest := range []echo.Instance{rc.A, rc.B, rc.Headless, rc.Multiversion, rc.Naked, rc.VM} {
```

Comment:
don't you need VM in src as well?

Reply:
The reason that I didn't put it in src is that VM does not have automatic DNS resolution. Therefore, I would need to hardcode [here](https://github.com/istio/istio/blob/767e291580462d924ea01cdd59bf882794135350/pkg/test/framework/components/echo/kube/deployment.go#L235) for the VM to connect to A, B, Multiversion, Naked, etc.. I could do this, but since #24919 is trying to improve the DNS resolution, I am kind of debating whether or not I should hard code the DNS resolution. I will think more about it and keep you updated. Thanks again for reviewing my code! The conversion has been helpful for me.

Reply:
I see, that makes sense now, I think this is part of the mesh expansion work and we shouldn't do more hack in the test for now, so just having VM in the dest is fine. thanks!

Reply:
Agreed, but let's make sure we do add it when DNS is ready

Reply:
@yangminzhu Since that PR is merged, I tried to add the VM to the src but it failed the ipv6 tests. I reverted back and I will be sure to add the VM to the src once the DNS with ipv6 problem is fixed. For now, I only put the VM in the dest. Hope this is acceptable for now!

Reply:
thanks!